test/e2e/gameserver_test.go

```diff
@@ -276,7 +289,10 @@ func TestGameServerRestartBeforeReadyCrash(t *testing.T) {
 
 			// create a connection each time, as weird stuff happens if the receiver isn't up and running.
 			conn, err := net.Dial("udp", address)
-			assert.NoError(t, err)
+			if !assert.NoError(t, err) {
```

Comment:
We should probably return the err here instead to escape out, and let the final check handle this, i think